### ISSUE TITLE
Cannot `/github subscribe` where there is a hyphen in the Github Org Name.

### ISSUE THREAD

**Issue by CallumShark** [integrations/slack#1120]
Definitely have the slack integration installed in github and the github app installed in slack, but it won't `/github subscribe SafeShark-Ltd/my-repo` 

- this happens where the Github **org name** has a hyphen in it.
- hyphens in the repo name do not affect it.
- I have multiple slack instances and multiple github accounts so have it working everywhere else...

`Only visible to you
21:30
Could not find resource: SafeShark-Ltd/my-repo.
Please check to make sure the spelling of the resource was correct and that you have access to the resource in GitHub.`

`Slack
 Installed 5 months ago  Developed by github  https://slack.github.com/`

**Comment by amitkumariiit** [comment]
Thanks @CallumShark for reaching out to us for the problem you are facing. Ideally this should not happen. Is the same org/repo `SafeShark-Ltd/my-repo` is able to be subscribed from other slack instances ? 
This error we get if we app doesn't have the permission for the repo and that happen if the github app is not installed in the repo. 
Can you just verify if app is installed for the `SafeShark-Ltd/my-repo` repo by going to the following settings
1. Go to org settings of org `SafeShark-Ltd`.
2. Go to `Installed GitHub Apps` settings.
3. See if you see the Slack app listed in the list of apps there. If Slack is there click on `configure` button and then on the landing page verify if the repo is selected on the `Repository access` section or the _All repositories_ section is selected.

**Comment by CallumShark** [comment]
Thanks for taking the time to look at this:

a) yes slack is configured on girhub for the SafeShark-Ltd org
`Installed GitHub Apps
GitHub Apps augment and extend your workflows on GitHub with commercial, open source, and homegrown tools.
Slack	Slack`

b) I clicked `configure` and the permissions were:

`Permissions`

`Read access to checks, code, commit statuses, metadata, and repository projects`

`Read and write access to deployments, issues, and pull requests`

and repo access is set to "all"

`Repository access`

`All repositories`

`This applies to all current and future repositories.`

I have other GitHub Orgs and other Slack instances and was able to (simply) get them integrated.  It is just this one that has the issues.  When I type `/github` on Slack, it gives me the normal options to subscribe etc.

Also, i have looked at the code in this repo.  I'm not an NPM expert,.but noticed that none of the test repo's has a repo with an org name that has a hyphen in it.  I was not confident to PR in a new  test.

**Comment by amitkumariiit** [comment]
Hi @CallumShark,
Are you able to subscribe to any other repo under the org `SafeShark-Ltd` ? 
Is another member able to subscribe ?

Can you try one thing. Try subscribing once after re-signning to github by running the `/github signin` command ? when prompted the github login page if for your org multi-factor authentication is enable, click authorise as shown in the below page. (this is for me, you will see another set up orgs)

If after that also it doesn't work, share the output of `/github debug` message with us for further debugging.
![image](https://user-images.githubusercontent.com/9694413/107499683-4542e380-6bbb-11eb-96f4-b811dcab250c.png)

**Comment by amitkumariiit** [comment]
if you don't see the above image once clicking on `connect github account` button because of the browser cache, try copping the link from the button by right clicking on it and open the url in the incognito window.

![image](https://user-images.githubusercontent.com/9694413/107501608-a4a1f300-6bbd-11eb-9133-f520dedd23ce.png)

**Comment by CallumShark** [comment]
Thankyou for the instructions.    

**resigning in:**

- I ran the `/github signin` cmd
- it says `Finish connecting your GitHub account`
- pressed the green button
- completed the re-oauth as per your image above
- ran '/github subscribe SafeShark-Ltd/one-of-my-repos` again
- **and it has worked**

OK, I know what's different here.

- I used chrome browser to do the oath rather than the slack client (for linux) this time.
- or, maybe the original setup did not work with the hyphen.
